src/VisualizationTests/HelixImageComparisonTests.cs

```diff
@@ -0,0 +1,428 @@
+            var bitmapsource = DynamoRenderBitmap(BackgroundPreview.View, 1024, 1024);
+#if UPDATEIMAGEDATA
+            UpdateTestData(path, bitmapsource);
+#endif
+            var refimage = new BitmapImage(new Uri(path));
+            var refbitmap = BitmapFromSource(refimage);
+            var newImage = BitmapFromSource(bitmapsource);
+
+            compareImageColors(refbitmap, newImage);
+        }
+
+        [Test]
+        public void linesSelected()
+        {
+            OpenVisualizationTest(@"imageComparison\linecolors.dyn");
+            RunCurrentModel();
+            var node1 = ViewModel.CurrentSpace.Nodes.Where(x => x.Name.Contains("coloredLines")).FirstOrDefault();
+            var node2 = ViewModel.CurrentSpace.Nodes.Where(x => x.Name.Contains("regularLines")).FirstOrDefault();
+            DynamoSelection.Instance.Selection.Add(node1);
+            DynamoSelection.Instance.Selection.Add(node2);
+            DispatcherUtil.DoEvents();
+
+            var path = generateTestDataPathFromTest(MethodBase.GetCurrentMethod().Name);
+            var bitmapsource = DynamoRenderBitmap(BackgroundPreview.View, 1024, 1024);
+#if UPDATEIMAGEDATA
+            UpdateTestData(path, bitmapsource);
+#endif
+            var refimage = new BitmapImage(new Uri(path));
+            var refbitmap = BitmapFromSource(refimage);
+            var newImage = BitmapFromSource(bitmapsource);
+
+            compareImageColors(refbitmap, newImage);
+        }
+
+        [Test]
+        public void linesFrozen()
+        {
+            OpenVisualizationTest(@"imageComparison\linecolors.dyn");
+            RunCurrentModel();
+            var node1 = ViewModel.CurrentSpace.Nodes.Where(x => x.Name.Contains("coloredLines")).FirstOrDefault();
+            var node2 = ViewModel.CurrentSpace.Nodes.Where(x => x.Name.Contains("regularLines")).FirstOrDefault();
+            node1.IsFrozen = true;
+            node2.IsFrozen = true;
+            DispatcherUtil.DoEvents();
+
+            var path = generateTestDataPathFromTest(MethodBase.GetCurrentMethod().Name);
+            var bitmapsource = DynamoRenderBitmap(BackgroundPreview.View, 1024, 1024);
+#if UPDATEIMAGEDATA
+            UpdateTestData(path, bitmapsource);
+#endif
+            var refimage = new BitmapImage(new Uri(path));
+            var refbitmap = BitmapFromSource(refimage);
+            var newImage = BitmapFromSource(bitmapsource);
+
+            compareImageColors(refbitmap, newImage);
+        }
+
+        #endregion
+
+        #region SpecialRenderPackages
+        [Test]
+        public void directManipulator()
+        {
+            var pointOriginNode =
+            new DSFunction(Model.LibraryServices.GetFunctionDescriptor("Point.ByCoordinates@double,double,double"));
+
+            var command = new Dynamo.Models.DynamoModel.CreateNodeCommand(pointOriginNode, 0, 0, true, false);
+            Model.ExecuteCommand(command);
+            RunCurrentModel();
```

Comment:
You could also create a dyn file with a Point node in it and add that node to the selection like you're doing here in order to trigger the creation of the manipulator.

Reply:
do you have a preference? I just felt it was less maintenance to do this - avoid another .dyn with a bunch of repeated json in it for one point.

Reply:
I don't have a preference. I made the note just for your info as an alternative to creating the manipulator in the test.